File: Week 02/id_181/590.n-ary-tree-postorder-traversal.js
/**
 * // Definition for a Node.
 * function Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */
/**
 * @param {Node} root
 * @return {number[]}
 */

// 后序遍历 根 左 右
var postorder = function(root) {
  const result = [];
  function pushRoot(root) {
      if(root.val===null) return result;    
      if(root.children.length>0) {
         for(var i = 0; i<root.children.length;i++) {
             pushRoot(root.children[i]);
         }
      }
      result.push(root.val)
  }
  root && pushRoot(root);
  return result;
};

File: Week 02/id_181/590.n-ary-tree-postorder-traversal.py
# [590. N叉树的后序遍历 - 力扣（LeetCode）](https://leetcode-cn.com/problems/n-ary-tree-postorder-traversal/)
class Solution:
    def postorder(self, root: 'Node'):
        res = []
        if root == None: return res
        stack = [root]
        
        while stack:
            curr = stack.pop()
            res.append(curr.val)
            stack.extend(curr.children)
            print(stack)
       
            
        return res[::-1]


# 前序排列后的倒序输出